runtime/src/main/cpp/Worker.cpp

```diff
@@ -234,16 +335,45 @@ class State {
     }
 
     Job job;
-    job.function = reinterpret_cast<KRef (*)(KRef, ObjHeader**)>(jobFunction);
-    job.argument = jobArgument;
-    job.future = future;
-    job.transferMode = transferMode;
+    if (jobFunction == nullptr) {
+      job.kind = JOB_TERMINATE;
+      job.terminationRequest.future = future;
+      job.terminationRequest.waitDelayed = !toFront;
+    } else {
+      job.kind = JOB_REGULAR;
+      job.regularJob.function = reinterpret_cast<KRef (*)(KRef, ObjHeader**)>(jobFunction);
+      job.regularJob.argument = jobArgument;
+      job.regularJob.future = future;
+      job.regularJob.transferMode = transferMode;
+    }
 
     worker->putJob(job, toFront);
 
     return future;
   }
 
+  bool executeJobAfterInWorkerUnlocked(KInt id, KRef operation, KLong afterMicroseconds) {
+    Worker* worker = nullptr;
+    {
+      Locker locker(&lock_);
+
+      auto it = workers_.find(id);
+      if (it == workers_.end()) return false;
+      worker = it->second;
+
+    }
+    Job job;
+    job.kind = JOB_EXECUTE_AFTER;
+    job.executeAfter.operation = CreateStablePointer(operation);
+    if (afterMicroseconds == 0) {
+      worker->putJob(job, false);
```

Comment:
Shouldn't the entire function body be executed under `lock_`? What happens if someone stops the worker before this line?